src/files/compose/validateDappnodeCompose.ts

```diff
@@ -0,0 +1,273 @@
+      if (cpServiceVolumes) {
+        for (const cpServiceVolume of cpServiceVolumes) {
+          validateComposeAndComposeServiceVolumes(
+            composeUnsafe,
+            isCore,
+            cpServiceName,
+            cpServiceVolume
+          );
+        }
+      }
+    }
+
+    if (aggregatedError.errors.length > 0) throw aggregatedError;
+  } catch (e) {
+    if (aggregatedError.errors.length > 0)
+      e.message += e.message + "\n" + aggregatedError.errors.join("\n");
+
+    throw e;
+  }
+}
+
+/**
+ * Ensures the docker compose version is supported
+ */
+function validateComposeVersion(compose: Compose): void {
+  if (
+    semver.lt(
+      compose.version + ".0",
+      params.MINIMUM_COMPOSE_FILE_VERSION + ".0"
+    )
+  )
+    aggregatedError.errors.push(
+      Error(
+        `Compose version ${compose.version} is not supported. Minimum version is ${params.MINIMUM_COMPOSE_FILE_VERSION}`
+      )
+    );
+}
+
+/**
+ * Ensures the docker compose networks are whitelisted
+ */
+function validateComposeNetworks(compose: Compose): void {
+  const cpNetworks = compose.networks;
+
+  if (cpNetworks) {
+    for (const networkName of Object.keys(cpNetworks)) {
+      // Check there are only defined whitelisted compose networks
+      if (!params.DOCKER_WHITELIST_NETWORKS.includes(networkName))
+        aggregatedError.errors.push(
+          Error(
+            `The docker network ${networkName} is not allowed. Only docker networks ${params.DOCKER_WHITELIST_NETWORKS.join(
+              ","
+            )} are allowed`
+          )
+        );
+      // Check all networks are external
+      if (cpNetworks[networkName].external === false)
+        aggregatedError.errors.push(
+          Error(
+            `The docker network ${networkName} is not allowed. Docker internal networks are not allowed`
+          )
+        );
+    }
+  }
+}
+
+/**
+ * Ensures the compose keys are whitelisted
+ */
+function validateComposeServiceKeys(
+  compose: Compose,
+  cpServiceName: string
+): void {
+  const cpServiceKeys = Object.keys(compose.services[cpServiceName]);
+  for (const cpServiceKey of cpServiceKeys) {
+    if (!params.SAFE_KEYS.includes(cpServiceKey))
+      aggregatedError.errors.push(
+        Error(
+          `Compose service ${cpServiceName} has key ${cpServiceKey} that is not allowed. Allowed keys are: ${params.SAFE_KEYS.join(
+            ","
+          )}`
+        )
+      );
+  }
+}
+
+/**
+ * Ensures the compose keys values are valid for dappnode
+ */
+function validateComposeServiceValues(
+  compose: Compose,
+  isCore: boolean,
+  cpServiceName: string
+): void {
+  // Check that if defined, the DNS must be the one provided from the bind package
+  const cpServiceDns = compose.services[cpServiceName].dns;
+  if (cpServiceDns && !params.DNS_SERVICE.includes(cpServiceDns))
+    aggregatedError.errors.push(
+      Error(
+        `Compose service ${cpServiceName} has DNS different than ${params.DNS_SERVICE}`
```

Comment:
Here we must force package to not specify any DNS

Reply:
I will add the restriction to be only declared by core packages so it is not a breaking change. Then we could start removing the `dns` from the core packages compose files

Reply:
Sounds good